ide-support/lsp/lsp-server/src/main/java/io/helidon/lsp/server/service/TextDocumentHandlerFactory.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.lsp.server.service;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import io.helidon.lsp.server.service.config.properties.PropertiesTextDocumentHandler;
+import io.helidon.lsp.server.service.config.yaml.YamlTextDocumentHandler;
+
+/**
+ * Factory for the TextDocumentHandler implementations.
+ */
+public class TextDocumentHandlerFactory {
+
+    private static final Map<String, TextDocumentHandler> FILE_EXTENSION_TO_HANDLER_MAP = new HashMap<>();
+    private static final TextDocumentHandler EMPTY_TEXT_DOCUMENT_HANDLER = new EmptyTextDocumentHandler();
+
+    private TextDocumentHandlerFactory() {
+    }
+
+    /**
+     * Get TextDocumentHandler instance for the file.
+     *
+     * @param fileName              File name.
+     * @return TextDocumentHandler instance for the file.
+     */
+    public static TextDocumentHandler   getByFileExtension(String fileName) {
+        final String fileExtension = getFileExtension(fileName);
+        if (fileExtension.equalsIgnoreCase("properties")) {
+            if (!FILE_EXTENSION_TO_HANDLER_MAP.containsKey("properties")) {
+                FILE_EXTENSION_TO_HANDLER_MAP.put("properties", PropertiesTextDocumentHandler.instance());
+            }
+        }
+        if (fileExtension.equalsIgnoreCase("yaml") || fileExtension.equalsIgnoreCase("yml")) {
+            if (!FILE_EXTENSION_TO_HANDLER_MAP.containsKey("yaml")) {
+                FILE_EXTENSION_TO_HANDLER_MAP.put("yaml", YamlTextDocumentHandler.instance());
+            }
+        }
+        return FILE_EXTENSION_TO_HANDLER_MAP.getOrDefault(fileExtension, EMPTY_TEXT_DOCUMENT_HANDLER);
+    }
+
+    /**
+     * Get file extension for the given file.
+     *
+     * @param fileName File name.
+     * @return File extension for the given file.
+     */
+    public static String getFileExtension(String fileName) {
```

Comment:
Can you use `FileUtils.fileExt` instead.

Reply:
In the method `getFileExtension` String parameter is used, method does not return null value and it converts result to lower case, so to substitube the method I have to do a few additional operations.